plutus-tx-plugin/src/PlutusTx/Options.hs

```diff
@@ -117,47 +117,71 @@ pluginOptions :: Map OptionKey PluginOption
 pluginOptions =
     Map.fromList
         [ let k = "typecheck"
-           in (k, PluginOption typeRep (setTrue k) posDoTypecheck mempty)
+              desc = "Do not perform type checking during compilation."
```

Comment:
Do _not_?

Reply:
Oops, wrote the description for `no-typecheck`.